packages/astro/src/core/config/schema.ts

```diff
@@ -525,6 +526,22 @@ export const AstroConfigSchema = z.object({
 				.boolean()
 				.optional()
 				.default(ASTRO_CONFIG_DEFAULTS.experimental.contentIntellisense),
+			responsiveImages: z
+				.union([
+					z.boolean(),
+					z.object({
+						layout: z
+							.union([
+								z.literal('responsive'),
+								z.literal('fixed'),
+								z.literal('full-width'),
+								z.literal('none'),
+							])
+							.optional(),
+					}),
+				])
+				.optional()
+				.default(ASTRO_CONFIG_DEFAULTS.experimental.responsiveImages),
```

Comment:
i wonder if you should transform it to always be an object?

Reply:
Good idea. That would simplify the types